### PR TITLE
fix(compaction): recover manual compaction with deterministic fallback

### PR DESCRIPTION
**Pull request by code-yeongyu**
## Summary
- Recover user-initiated manual compaction through the existing deterministic no-LLM fallback for classified summarizer failures.
- Preserve fail-closed behavior for non-required routes and unclassified failures.

## Root cause
The core `session_before_compact` recovery gate only treated `threshold` and `overflow` as required routes. Manual `/compact` uses `reason: "manual"`, so classified summarizer failures propagated as cancellation instead of using the already-implemented deterministic recovery.

## Fix
- Include `manual` in `isRequiredCompactionFallbackReason`.
- Add handler-level tests for empty-summary and watchdog timeout recovery, non-required fail-closed behavior, and manual auth failure.
- Update the compaction tracker and coding-agent changelog.

## Tests
RED before production change:
```
2 failed | 7 passed (9)
manual empty-summary and manual timeout returned { cancel: true } instead of a deterministic compaction
```

GREEN after production change:
```
Test Files  1 passed (1)
Tests  9 passed (9)
```

Additional verification:
- Local compaction suite: 74 files, 533 tests passed.
- Biome: passed.
- Changelog gate: passed.
- Remote compaction suite: 72 passed, 2 pre-existing timeout failures in `blocking-compaction-review-hardening.test.ts` and `blocking-compaction-shared-retry.test.ts` (reproduced on rerun; changed tests all pass).

## Behavior change note
Manual `/compact` now recovers context deterministically for classified summarization failures (`summarization-timeout`, `upstream-stream-truncated`, `summarization-overflow-exhausted`, and `summarization-empty-summary`). Auth and other unclassified errors still fail closed; speculative/idle and other non-required routes are unchanged.

---
## Summary by cubic
Manual `/compact` previously cancelled on classified summarizer failures; it now uses the existing deterministic no-LLM fallback. Unclassified failures, including auth errors, and non-required routes still fail closed, while successful recovery does not record a breaker failure.

- Covers timeout, truncated-stream, overflow-exhausted, and empty-summary failures.
- Adds tests for manual recovery, fail-closed cases, and breaker bookkeeping.
- Documents the policy in the compaction changes log and `packages/coding-agent` changelog.

<sup>Written for commit bf42495296b325434c823e5c5d922683f781220e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/code-yeongyu/senpi/pull/1273?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/coding-agent/test/compaction/required-compaction-deterministic-fallback.test.ts [hunk 1]
```diff
@@ -133,7 +133,6 @@ describe("required compaction deterministic fallback", () => {
 
 	it("does not recover manual, aborted, or unrelated failures", async () => {
 		for (const testCase of [
-			{ reason: "manual" as const, message: "upstream_stream_truncated", aborted: false, refusal: false },
```
[the remark this answers is not in the document]
Comment (code-yeongyu) [reply packages/coding-agent/test/compaction/required-compaction-deterministic-fallback.test.ts:136]: Addressed in 641431856: retitled to 'does not recover aborted or unrelated failures' - manual now recovers via the deterministic fallback by design (the manual + unclassified auth case still pins fail-closed).
